src/Reflection/Constant/RuntimeConstantReflection.php

```diff
@@ -2,17 +2,28 @@
 
 namespace PHPStan\Reflection\Constant;
 
+use PHPStan\PhpDoc\ResolvedPhpDocBlock;
 use PHPStan\Reflection\GlobalConstantReflection;
 use PHPStan\TrinaryLogic;
+use PHPStan\Type\FileTypeMapper;
 use PHPStan\Type\Type;
 
 class RuntimeConstantReflection implements GlobalConstantReflection
 {
 
+	private false|ResolvedPhpDocBlock $resolvedPhpDocBlock = false;
+
+	private ?string $deprecatedDescription = null;
+
+	/**
+	 * @param non-empty-string|null $docComment
+	 */
 	public function __construct(
 		private string $name,
 		private Type $valueType,
 		private ?string $fileName,
+		private FileTypeMapper $fileTypeMapper,
```

Comment:
I'd rather be if this constructor only accepted `bool $isDeprecated` and `?string $deprecatedDescription`. They should be resolved where this object is created.